crates/miden-objects/src/account/mod.rs

```diff
@@ -389,6 +381,14 @@ impl Account {
             });
         }
 
+        // Maintain internal consistency of the account, i.e. the seed should not be present for
+        // existing accounts, where existing accounts are defined as having a nonce > 0.
+        // If we've incremented the nonce, then we should remove the seed (if it was present at
+        // all).
+        if new_nonce.as_int() > 0 {
```

Comment:
very minor nit: Should this be checked via `Account::is_new()`? The definition of "new" accounts seems unlikely to change but if it ever does (and while "we need seeds for new accounts" still holds) the check would still be valid.


```suggestion
        if !self.is_new() {
```